### PR TITLE
Feature/authenticate

### PR DESCRIPTION
**Pull request by johandroid**

### CODE REVIEW

**Review by jgutierrezre — COMMENTED** [review]
lgtm

**Review by jgutierrezre — APPROVED** [review]
lgtm

File: pallets/pass/src/lib.rs [hunk 1]
```diff
@@ -441,18 +426,18 @@ pub mod pallet {
...
-            // ensure!(
-            //     auth_method::authenticate(device_id, challenge_signature, block_number_mod_10_bytes).is_ok(),
-            //     Error::<T, I>::ChallengeFailed
-            // );
-            // </Validate device>
+            // Verify signature of device
+            auth_method
+                .authenticate(
+                    device_id.to_vec(),
+                    &block_number_mod_x_bytes[..],
```
Comment (pandres95) [inline pallets/pass/src/lib.rs:433]: Use the `T::Randomness` to preserve consistency across methods

**Review by pandres95 — CHANGES_REQUESTED** [review]
Overall is good. Just consider we need to use `Randomness` on every step we attempt an authentication.

**Review by pandres95 — APPROVED** [review]

**Pull request merged**